volatility3/framework/symbols/linux/utilities/modules.py

```diff
@@ -116,3 +165,369 @@ def lookup_module_address(
                 break
 
         return mod_name, symbol_name
+
+    @classmethod
+    def get_module_info_for_module(
+        cls, address_mask: int, module: extensions.module
+    ) -> Optional[ModuleInfo]:
+        """
+        Returns a ModuleInfo instance for `module`
+
+        This performs address masking to avoid endless calls to `mask_mods_list`
+
+        Returns None if the name is smeared
+        """
+        try:
+            mod_name = utility.array_to_string(module.name)
+        except exceptions.InvalidAddressException:
+            return None
+
+        start = module.get_module_base() & address_mask
+
+        end = start + module.get_core_size()
+
+        return Modules.ModuleInfo(module.vol.offset, mod_name, start, end)
+
+    @staticmethod
+    def get_kernel_module_info(
+        context: interfaces.context.ContextInterface,
+        kernel_module_name: str,
+    ) -> ModuleInfo:
+        """
+        Returns a ModuleInfo instance that encodes the kernel
+        This is required to map function pointers to the kerenl executable
+        """
+        kernel = context.modules[kernel_module_name]
+
+        mask = context.layers[kernel.layer_name].address_mask
+
+        start_addr = kernel.object_from_symbol("_text")
+        start_addr = start_addr.vol.offset & mask
+
+        end_addr = kernel.object_from_symbol("_etext")
+        end_addr = end_addr.vol.offset & mask
```

Comment:
```suggestion
        address_mask = context.layers[kernel.layer_name].address_mask

        start_addr = kernel.object_from_symbol("_text")
        start_addr = start_addr.vol.offset & address_mask

        end_addr = kernel.object_from_symbol("_etext")
        end_addr = end_addr.vol.offset & address_mask
```
Could we call this address_mask to keep it consistent with other methods please?

Reply:
changed